temporian/implementation/numpy_cc/operators/calendar.cc

```diff
@@ -0,0 +1,243 @@
+#include <pybind11/iostream.h>
+#include <pybind11/numpy.h>
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+
+#include <iostream>
+
+#include "absl/time/time.h"
+#include "temporian/implementation/numpy_cc/operators/common.h"
+
+namespace {
+namespace py = pybind11;
+
+std::optional<std::string> parse_tz(const py::object &tz, absl::TimeZone &parsed_tz) {
+  if (py::isinstance<py::int_>(tz)) {
+    // handle tz as an int
+    const int int_tz = tz.cast<int>();
+    // tz is expresed in hours, needs to be converted to seconds
+    parsed_tz = absl::FixedTimeZone(int_tz * 60 * 60);
+
+  } else if (py::isinstance<py::float_>(tz)) {
+    // handle tz as an float
+    const float float_tz = tz.cast<float>() * 60 * 60;
+    const int int_tz = static_cast<int>(std::round(float_tz));
+    // tz is expresed in hours, needs to be converted to seconds
+    parsed_tz = absl::FixedTimeZone(int_tz);
+
+  } else if (py::isinstance<py::str>(tz)) {
+    // handle tz as an string
+    const std::string str_tz = tz.cast<std::string>();
+    if (!absl::LoadTimeZone(str_tz, &parsed_tz)) {
+      return "Invalid timezone '" + str_tz +
+             "'. Only names defined in the IANA timezone" +
+             "database are valid";
+    }
+
+  } else {
+    return "Unsupported argument type for \"tz\" argument. Expecting int, "
+           "float, "
+           "or str";
+  }
+
+  return {};
+}
+
+int is_leapyear(int64_t year) {
```

Comment:
Use temporian/implementation/numpy_cc/operators/tick_calendar_utils.cc

Also, add new functions in tick_calendar_utils.cc and add unit tests

Reply:
Makes sense, will switch to `IsLeapYear` and move `map_week_day` and `parse_tz` to the utils file. I have a couple of notes though:

1. I think it'd be a good idea to rename `tick_calendar_utils` to `calendar_utils` since now they are not just for tick operations.
2. I notice that you are using PascalCase for the functions but I was using snake_case, is there a preference in cpp for it? I tend to default to snake_case automatically because of Python